runtime/near-vm-runner/src/tests/error_cases.rs

```diff
@@ -130,25 +131,22 @@ fn trap_contract() -> Vec<u8> {
 
 #[test]
 fn test_trap_contract() {
-    // See the comment is test_stack_overflow.
-    with_vm_variants(|vm_kind: VMKind| match vm_kind {
-        VMKind::Wasmtime => return,
-        VMKind::Wasmer0 => assert_eq!(
-            make_simple_contract_call_vm(&trap_contract(), "hello", vm_kind),
-            (
-                Some(vm_outcome_with_gas(47105334)),
-                Some(VMError::FunctionCallError(FunctionCallError::WasmUnknownError))
-            )
-        ),
-        VMKind::Wasmer1 => assert_eq!(
+    with_vm_variants(|vm_kind: VMKind| {
+        match vm_kind {
+            VMKind::Wasmer0 | VMKind::Wasmer1 => {}
+            // All contracts leading to hardware traps can not run concurrently on Wasmtime and Wasmer,
+            // Restore, once get rid of Wasmer 0.x.
+            VMKind::Wasmtime => return,
+        }
+        assert_eq!(
             make_simple_contract_call_vm(&trap_contract(), "hello", vm_kind),
             (
                 Some(vm_outcome_with_gas(47105334)),
-                Some(VMError::FunctionCallError(FunctionCallError::Wasmer1Trap(
-                    "unreachable".to_string()
+                Some(VMError::FunctionCallError(FunctionCallError::WasmTrap(
+                    WasmTrap::Unreachable
```

Comment:
This test returns WasmUnknownError for wasmer 0.x, did you change somewhere in code to make it return unreachable too?